EmployeeController.java

```diff
@@ -29,38 +35,43 @@ public class EmployeeController {
     private EmployeeRepository employeeData;
 
     @RequestMapping(value = "/addNewEmployee", method = RequestMethod.POST)
-    public Employee addEmployee(@RequestBody Employee employee) {
+    public ResponseEntity<Object> addEmployee(@RequestBody Employee employee) {
 //get data from each variable and do validations
         EmployeeValidator validator = new EmployeeValidator();
-        if(validator.isPhoneNumberValid(employee.getPhoneNumber()))
-        return employeeData.save(employee);
+        if(validator.isPhoneNumberValid(employee.getPhoneNumber()) && validator.isDojValid(employee.getDateOfJoining()) && validator.isEmailValid(employee.getEmail())) {
+            employeeData.save(employee);
+            return new ResponseEntity<Object>("Employee successfully added in DB",HttpStatus.OK);
+        }
         else
-            throw new InvalidEmployeeDataException("invalid phone number entered");
-
+        return new ResponseEntity<Object>(new InvalidEmployeeDataException("invalid data entered"), HttpStatus.OK);
     }
 
     @RequestMapping(value = "/listEmployees", method = RequestMethod.GET)
-    public List<EmployeeTax> getEmployeesTaxInfo() {
-        List<Employee> allEmployees = (List<Employee>) employeeData.findAll().iterator();
+    public ResponseEntity<Object> getEmployeesTaxInfo() {
+        List<Employee> allEmployees = ((ArrayList<Employee>) employeeData.findAll());
         List<EmployeeTax> taxedEmployees = new ArrayList<>();
 
         for (Employee employee:allEmployees) {
             EmployeeTax taxData = null;

```

Comment:
you could have tried model mapper or map struct for mapping

Reply:
Agreed. Though I used these in my very few previous projects, currently we are not using it in my project due to below reasons. 
Model mappers relies on reflection which slows down and affects performance as project size increases. With mapStruct, projects with large number of mappers increase the size of generated code and it won’t support all third party libraries.

So my client was not interested to use these, so I am not thought of using it here. This is small project so used manual mapping. Thanks